engine/servermaster/server.go

```diff
@@ -298,30 +299,78 @@ func (s *Server) DeleteJob(ctx context.Context, req *pb.DeleteJobRequest) (*empt
 }
 
 // RegisterExecutor implements grpc interface, and passes request onto executor manager.
-func (s *Server) RegisterExecutor(ctx context.Context, req *pb.RegisterExecutorRequest) (*pb.RegisterExecutorResponse, error) {
-	resp2 := &pb.RegisterExecutorResponse{}
-	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp2)
+func (s *Server) RegisterExecutor(ctx context.Context, req *pb.RegisterExecutorRequest) (*pb.Executor, error) {
+	executor := &pb.Executor{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &executor)
 	if shouldRet {
 		if err != nil {
 			log.Warn("RegisterExecutor failed", zap.Error(err))
 			return nil, err
 		}
-		return resp2, err
+		return executor, err
 	}
-	// register executor to scheduler
-	// TODO: check leader, if not leader, return notLeader error.
-	execInfo, err := s.executorManager.AllocateNewExec(req)
+
+	nodeInfo, err := s.executorManager.AllocateNewExec(req)
 	if err != nil {
-		log.Error("add executor failed", zap.Error(err))
-		return &pb.RegisterExecutorResponse{
-			Err: cerrors.ToPBError(err),
-		}, nil
+		log.Error("register executor failed", zap.Error(err))
+		return nil, err
 	}
-	return &pb.RegisterExecutorResponse{
-		ExecutorId: string(execInfo.ID),
+	return &pb.Executor{
+		Id:         string(nodeInfo.ID),
+		Name:       nodeInfo.Name,
+		Address:    nodeInfo.Addr,
+		Capability: int64(nodeInfo.Capability),
 	}, nil
 }
 
+// ListExecutors implements DiscoveryServer.ListMasters.
+func (s *Server) ListExecutors(ctx context.Context, req *pb.ListExecutorsRequest) (*pb.ListExecutorsResponse, error) {
+	resp := &pb.ListExecutorsResponse{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp)
+	if shouldRet {
+		if err != nil {
+			log.Warn("ListExecutors failed", zap.Error(err))
+			return nil, err
+		}
```

Comment:
I moved log to `rpcerror.UnaryServerInterceptor` so that we don't need to log in every method. https://github.com/pingcap/tiflow/pull/6864/commits/1682ab205015acdb5fe3ae1a2638abbea522dd6c